### ISSUE TITLE
virtualenv-2.7 & virtualenv-3.4 are removed, which breaks venv.sh

### ISSUE THREAD

**Issue by yusiwen** [rafi/vim-config#42]
See pypa/virtualenv#692

**Comment by rafi** [comment]
That's actually links that Macports creates when installing virtualenv.
Notice in `venv.sh` I'm also trying `virtualenv2`and `virtualenv3`...
How are yours called?

**Comment by randavidovitz** [comment]
i simply ran 
`pip3 install -U neovim PyYAML`

**Comment by rafi** [comment]
`venv.sh` tries to look for:
* virtualenv3 or virtualenv-3.5
* virtualenv2 or virtualenv-2.7

You have to have `virtualenv` installed, both python2 and python3 version.

**Comment by gfreivasc** [comment]
In my linux installation, there is no difference between virtualenv versions. It actually creates environments that support both `python` and `python3` (there is no `python2`, it's only `python`). Both `pip` and `pip3` installations point to the same package and will alert that it is already installed when I try to run it on the second pip.

**Comment by olymk2** [comment]
same i have virtualenv installed but there are no numbers prefixed, so it would be nice if the script checked for that to be more complete.
Don't really want to pip install virtualenv when i already have it installed, especially as I don't really use virtualenv's any more and just use docker.

**Comment by rafi** [comment]
@gfreivasc @olymk2, do you use `pyenv` or `virtualenv`? How do you create py2 and py3 envs?

The script tries to create 2 virtual-environments and have Neovim use them as scripted [here](https://github.com/rafi/vim-config/blob/master/config/neovim.vim#L4-L10).  _Virtual-environments are NOT a must!_ Notice that config/neovim.vim first checks if those directories exist.  Otherwise, if you don't want to use virtual-environments you can just use `pip install neovim` to install Neovim python bindings globally.

So to summarize:
1. This is only for Neovim features, if you use classic Vim you don't need this.
1. You don't have to use virtual-environments, read `:h provider-python`
1. I'd love suggestions and examples on how you guys create python2 and python3 virtual-environments so I could amend the setup script. (My workstation is macOS with Macports)

**Comment by olymk2** [comment]
I don't really use virtualenv but I install it using the system package manager command below.

    apt install python-virtualenv

The issue is more that the neovim setup script stops or breaks when virtualenv is not found so if it checked and skipped that part, or asked if you wanted it that would be helpful.

I really used this only because i wanted a quick vim setup for editing scripts on a raspberry pi, tend to use spacemacs but its huge to do a quick install on a pi so something pre configured for editing python would be ideal.

**Comment by rafi** [comment]
`venv.sh` has since been updated. Let me know if there's still issues..
